locale/en/locale.po

```diff
@@ -1415,6 +1415,9 @@ msgstr "Search articles for"
 msgid "search.withinJournal"
 msgstr "Within"
 
+msgid "search.journal"
```

Comment:
Adding this to `stable-3_4_0` is going to cause an untranslated locale key to appear unexpectedly on pretty much all non-English translations. We usually have two approaches to this...
- Merge the translation addition to `stable-3_4_0` some months ahead of time to give translators a chance to complete it, then merge the code change later, or
- Fill in best-guess translations with a `fuzzy` flag based on machine translation, at least for the "critical" translations (French, Portuguese, German, Spanish, etc)

Reply:
Ok, I decided to remove the new locale key and use just "Journal" as the field label. I think that's descriptive enough for `stable-3_4_0`. I'll put the "By Journal" locale key into the `main` branch.